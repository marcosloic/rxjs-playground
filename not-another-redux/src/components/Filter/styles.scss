.filter {
    padding: 10px;
    .title {
        text-transform: capitalize;
        margin-bottom: 5px;
    }
}
