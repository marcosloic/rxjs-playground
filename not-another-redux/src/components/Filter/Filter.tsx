import React from 'react';
import { FiltersState } from '../../models/filters';
import './styles.scss';

type FilterPropTypes = {
    title: keyof FiltersState;
    values: FiltersState[keyof FiltersState];
    onChangeCallback: ({ title, value }) => void;
};

export const Filter: React.FunctionComponent<FilterPropTypes> = React.memo(
    props => {
        const { title, values, onChangeCallback } = props;
        const valuesArray = Array.from(values);
        const onValueChange = React.useCallback(
            event => {
                onChangeCallback({ title, value: event.target.value });
            },
            [title, onChangeCallback]
        );
        const defaultSelectValue = `Pick some ${title}`;
        return (
            <div className="filter">
                <h5 className={'title'}>{title}</h5>
                <select onChange={onValueChange} defaultValue={'default'}>
                    <option disabled value={'default'}>
                        {defaultSelectValue}
                    </option>
                    {valuesArray.sort().map(value => (
                        <option key={value} value={value}>
                            {value}
                        </option>
                    ))}
                </select>
            </div>
        );
    }
);

Filter.defaultProps = {
    title: '',
    values: new Set(),
    onChangeCallback: () => {}
};

Filter.displayName = 'FiltersList';
