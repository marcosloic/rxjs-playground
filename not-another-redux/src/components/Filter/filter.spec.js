import TestRenderer from 'react-test-renderer';
import React from 'react';
import { Filter } from './Filter';

describe('The filter component', () => {
    it('Matches the snapshot', () => {
        const filter = TestRenderer.create(<Filter />);
        expect(filter).toMatchSnapshot();
    });
});
