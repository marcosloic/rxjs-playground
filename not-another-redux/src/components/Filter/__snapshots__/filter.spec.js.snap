// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The filter component Matches the snapshot 1`] = `
<div
  className="filter"
>
  <h5
    className="title"
  >
    
  </h5>
  <select
    defaultValue="default"
    onChange={[Function]}
  >
    <option
      disabled={true}
      value="default"
    >
      Pick some 
    </option>
  </select>
</div>
`;
